study/hello-caqti-eio/README.md

```diff
@@ -0,0 +1,51 @@
+# Hello, caqti-eio!
+
+Let's look at [eio](https://github.com/ocaml-multicore/eio)!
+
+Instead of promises, we will instead use a "direct style" concurrency model.
+
+> NOTE: OCaml v5 or greater is required.
+
+As always let's run our tests:
+
+```
+cd ./hello-caqti-eio
+PGHOST=localhost PGDATABASE=caqti_study PGPORT=5433 dune exec ./bin/main.exe
+PGHOST=localhost PGDATABASE=caqti_study PGPORT=5433 dune runtest --watch
+```
+
+## Test via the REPL
+
+```
+$ PGHOST=localhost PGDATABASE=caqti_study PGPORT=5433 dune utop
+```
+
+Firstly, let's bring the `Init` and `Exec` modules into scope:
+```ocaml
+# open Repo
+```
+
+Also, we will create this helper function, for demonstration purposes.
+
+```ocaml
+# let with_conn f = Result.get_ok @@ Eio_main.run @@ Init.with_conn @@ f;;
```

Comment:
Using `Caqti_eio.or_fail` will give a more precise exception, and maybe commute into eio-main.